assembly/pallets/transaction-payment/payment.ts

```diff
@@ -0,0 +1,25 @@
+import { DispatchInfo, ExistenceRequirement, WithdrawReasons } from "subsembly-core";
+import { AccountIdType, Balance, Weight } from "../../runtime/runtime";
+import { Balances } from "../balances/balances";
+
+/**
+ * @description Handle withdrawing, refunding and depositing of transaction fees.
+ */
+export class Payment {
```

Comment:
`Payment` and `TransactionPayment` seem very redundant to me (it terms of names). I think that we could rename `Transaction-Payment` to `FeeCalculator`